zsh-aws-vault.plugin.zsh

```diff
@@ -35,6 +37,61 @@ function avsh() {
 }
 
 function avli() {
+  function _using_osx() {
+    [[ $(uname) == "Darwin" ]]
+  }
+
+  function _using_linux() {
+    [[ $(uname) == "Linux" ]]
+  }
+
+  function _find_browser() {
+    if [ -n "${AWS_VAULT_PL_BROWSER}" ]; then
+      # use the browser bundle specified
+      echo "${AWS_VAULT_PL_BROWSER}"
+    elif [ -n "${BROWSER}" ]; then
+        echo "${BROWSER}"
+    elif _using_osx ; then
+      # Detect the browser in launchservices
+      # https://stackoverflow.com/a/32465364/808678
+      local prefs=~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure.plist
+      plutil -convert xml1 $prefs
+      grep 'https' -b3 $prefs | awk 'NR==2 {split($2, arr, "[><]"); print arr[3]}';
+      plutil -convert binary1 $prefs
+    elif _using_linux ; then
+      # This is bad but it's marginally better than hardcoding google-chrome
+      xdg-settings get default-web-browser | cut -d'.' -f1
+    else
+      # TODO - other platforms
+    fi
+  }
+
+  function _get_browser_profile_path() {
+    local browser=$1
+    local profile=$2
+    local browser_profile_path=""
+
+    if [ "$AWS_VAULT_PL_PERSIST_PROFILE" = "true" ]; then
+      browser_profile_path="${AWS_VAULT_PL_PERSIST_PROFILE_PATH}/${browser}/${profile}"
+      mkdir -p "${AWS_VAULT_PL_PERSIST_PROFILE_PATH}/${browser}"
+    else
+      browser_profile_path=$(mktemp --tmpdir -d $browser.$profile.XXXXXX)
+    fi
+
+    echo $browser_profile_path
+  }
+
+  function _maybe_clean_up_browser_profile() {
+    local browser_profile_path=$1
+    if [ "$AWS_VAULT_PL_PERSIST_PROFILE" = "false" ]; then
+      rm -rf $browser_profile_path
+    fi
+  }
+
+  function _chromium_common_flags() {
+    echo "--no-first-run --new-window"
```

Comment:
In the scenario where I'm persisting profiles and my session times out; I just call the `avli {profile}` command again which launches a new window. Since the new window and already opened window share the same browser profile both have the refreshed AWS session.  
In this scenario I've essentially set the `--new-window` to only get added when I'm not using the persist profile option.

Reply:
Got it - that makes sense. In reviewing this, I think we shouldn't set `--no-first-run` _ever_ by default. If people want this behavior, they can set it via `AWS_VAULT_PL_BROWSER_LAUNCH_OPTS`.

I'll add a note in the README about this.